### PR TITLE
small perfromance fixes

### PR DESCRIPTION
**Pull request by tomerd**
motivation: better performance

changes:
* cache manifest dependencies and target computations
* change several functions that return static data to computed properties

### CODE REVIEW

**Comment by tomerd** [comment]
@swift-ci smoke test

File: Sources/PackageModel/Manifest/Manifest.swift [hunk 1]
```diff
@@ -304,17 +325,18 @@ public final class Manifest: Sendable {
...
         }
 
-        return packageDependency(referencedBy: packageName)
+        return self.packageDependency(referencedBy: packageName)
     }
 
     /// Finds the package dependency referenced by the specified plugin usage.
-    /// - Returns: Returns `nil` if  the used plugin is from the same package or if the package the used plugin is from cannot be found.
+    /// - Returns: Returns `nil` if  the used plugin is from the same package or if the package the used plugin is from
+    /// cannot be found.
```
Comment (neonichu) [inline Sources/PackageModel/Manifest/Manifest.swift:333]: 😞

File: Sources/Workspace/Workspace.swift [hunk 2]
```diff
@@ -1878,6 +1878,8 @@ extension Workspace {
             // FIXME: This should be an ordered set.
             requiredIdentities = inputIdentities.union(requiredIdentities)
 
+
+
```
Comment (neonichu) [inline Sources/Workspace/Workspace.swift:1882]: 🤔

**Comment by tomerd** [comment]
@swift-ci smoke test

**Review by neonichu — APPROVED** [review]

**Pull request merged**